dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -149,111 +152,112 @@ public void removeActionlet(String workFlowActionletName) {
-		return wfac.findStepByContentlet(contentlet);
+		return workFlowFactory.findStepByContentlet(contentlet);
 	}
 
 	public WorkflowTask findTaskById(String id) throws DotDataException {
-		return wfac.findWorkFlowTaskById(id);
+		return workFlowFactory.findWorkFlowTaskById(id);
 	}
 
-	public List<WorkflowScheme> findSchemes(boolean showArchived) throws DotDataException {
-		return wfac.findSchemes(showArchived);
+	@CloseDBIfOpened
+	public List<WorkflowScheme> findSchemes(final boolean showArchived) throws DotDataException {
+		return workFlowFactory.findSchemes(showArchived);
 	}
 
+	@CloseDBIfOpened
 	public WorkflowScheme findDefaultScheme() throws DotDataException {
-		return wfac.findDefaultScheme();
+		return workFlowFactory.findDefaultScheme();
 	}
 
+	@CloseDBIfOpened
 	public boolean isDefaultScheme(WorkflowScheme scheme) throws DotDataException {
 		if (scheme == null || scheme.getId() == null) {
 			return false;
 		}
-		if (wfac.findDefaultScheme().getId().equals(scheme.getId())) {
+		if (workFlowFactory.findDefaultScheme().getId().equals(scheme.getId())) {
 			return true;
 		}
 		return false;
 	}
 
+	@CloseDBIfOpened
 	public WorkflowScheme findScheme(String id) throws DotDataException {
-		return wfac.findScheme(id);
+		return workFlowFactory.findScheme(id);
 	}
-	public void saveSchemeForStruct(Structure struc, WorkflowScheme scheme) throws DotDataException {
 
-		try{
-			wfac.saveSchemeForStruct(struc.getInode(), scheme);
-		}
-		catch(DotDataException e){
+	@WrapInTransaction
+	public void saveSchemeForStruct(final Structure struc, final WorkflowScheme scheme) throws DotDataException {
+
+		try {
+			workFlowFactory.saveSchemeForStruct(struc.getInode(), scheme);
+		} catch(DotDataException e){
 			throw e;
 		}
 	}
-	public WorkflowScheme findSchemeForStruct(Structure struct) throws DotDataException {
+
+	@CloseDBIfOpened
+	public WorkflowScheme findSchemeForStruct(final Structure structure) throws DotDataException {
 
 
-		if(struct ==null || ! UtilMethods.isSet(struct.getInode()) || LicenseUtil.getLevel() < LicenseLevel.STANDARD.level){
+		if(structure ==null || ! UtilMethods.isSet(structure.getInode()) || LicenseUtil.getLevel() < LicenseLevel.STANDARD.level){
 			return findDefaultScheme();
 		}
 		try{
-			return wfac.findSchemeForStruct(struct.getInode());
+			return workFlowFactory.findSchemeForStruct(structure.getInode());
 		}
 		catch(Exception e){
 			return findDefaultScheme();
 		}
-
-
-
-
-
-
 	}
 
-	public void saveScheme(WorkflowScheme scheme) throws DotDataException, AlreadyExistException {
+	@WrapInTransaction
+	public void saveScheme(final WorkflowScheme scheme) throws DotDataException, AlreadyExistException {
 		
-		wfac.saveScheme(scheme);
+		workFlowFactory.saveScheme(scheme);
 
 	}
 
 	public void deleteScheme(WorkflowScheme scheme) throws DotDataException {
-		// TODO Auto-generated method stub
 
 	}
 
 	public void activateScheme(WorkflowScheme scheme) throws DotDataException {

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)